### DISCUSSION TITLE
Welcome to nativescript-vue discussions!

### DISCUSSION THREAD

**Discussion by rigor789** [nativescript-vue/nativescript-vue#764] [General]
## 👋 Welcome!
  We’re now using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with NativeScript and Vue!

**Comment by rigor789** [comment]
Let me start! I've started using NativeScript in early 2017. About 2 days into that journey I started writing nativescript-vue... It's been a long journey but definitely taught me how it all works!

The majority of features I work on come from using nativescript-vue for client applications and discovering less intuitive areas I want to improve!

**Comment by tralves** [comment]
Hi! I am Tiago!

I started playing with NS-Vue in 2017 too. Then Progress asked me to build the groceries demo app in ns-vue, yada-yada-yada, I am a fulltime NS-Vue developer at IRC.com!

**Comment by kaanguru** [comment]
I am Cem Kaan
I've built my first project in 2020. I like Vue. I believe nativescript-vue has a strong future.
NativeScript Community is formed by helpful individuals.  

I believe all users must do whatever they can to support. I would like to contribute a few days on some _good first issue_ tagged issues.

## Some Possible Future Features

- **Hot Reload**
- Better Typescript definitions
- Unit Testing
- Components documentation updates for NS 7
- Vue 3 Support

## How can users contribute?

- Bug reporting and bugfixes in several scales. 
- Contribute to some plugins. 
- Write some tutorials.
- Write some **Recipes** that demonstrate how to solve common problems.
- Answer [questions](https://stackoverflow.com/questions/tagged/nativescript-vue).
- [Support developers](https://www.patreon.com/rigor789)

**Comment by ghost** [comment]
Hello ! I am Fabrice Iyongo.
I would also have liked to add if possible to have a discussion forum on the www.nativescript-vue.org site, so I would like the site to be updated often with lots of articles on his blog regarding nativescript and nativescript-vue. For the moment I have the impression that the project is on hiatus is abandoned, without forgetting a better support for typescript, hot reloading and especially I would like to know when the official release of nativescript-vue 3.0, since its announcement I was patient to the point that I became impatient.

I'm happy about the idea of having a community behind nativescript-vue with a place to exchange ideas, questions and improve. Strongly a French-speaking community, I have a lot of trouble with English

**Comment by Rho-bur** [comment]
Hi,
I am Robertino Vasilescu.
I am a backend developer for many years and started to use NS plus Vue since 2019.
I find it extremely helpful that I can reuse my Vue components built initially for web ans I am now totally sold out to NS-Vue.
New Features:
- maybe Vue-Router?

Regards,

**Comment by vallemar** [comment]
Hi,
I use Vue for my web applications, and of course, for my mobile applications I use Nativescript-Vue. I have been using Nativescript-Vue since 2019 and the truth is that I am delighted, I would like to have vue 3 here since the change regarding the code organization is great.

**Comment by baybal** [comment]
I'm not as delighted as vallemar.

7.0 been a trainwreck, and 6.0 was a performance/memory leak nightmare.

While terrible upstream functionality is upstream's problem, I'm continuing to see issues more, or less along the same lines in nativescript-vue.

I spent 2 days worth of work hours trying to debug why a modal is not closing upon $modal.close, and only does so after calling $modal.close multiple times. I'm still trying to figure it out.

Debug output is terrible, and typescript gibberish + bable + webpack garbling stack traces adds to the problem.

Finding out what is the root cause of an exception inside a Vue instance is really hard, and takes hours work in the debugger (which is itself half broken on nativescript 7.0)

In comparison to just using notepad, and vanilla Java for Android development, nativescript in its current state is worse, much worse. I very much regret going for it in a large app project, but can't get out of the trap now :(.

**Reply by baybal** [reply]
I'd say, as it is now, using Nativescript ecosystem certainly doesn't save _any_ time over developing two separate apps with native tools, and in my case it already doubled it.

**Reply by rigor789** [reply]
Thanks for sharing your honest experience, I will not try to (nor do I want to) dismiss the pain points you have. 

I've experienced some of them over the past few years, but nothing that made me want to switch the tech stack. But I do understand my experience is different - as I've built NativeScript-Vue, and am now part of the core team for NativeScript itself. I've been focusing on improving the DX, improving the CLI & rewriting the Webpack integration (still WIP, but mostly functional at this point, [see RFC](https://github.com/NativeScript/rfcs/pull/2)), simplifying things where I can etc. 

I've always been open to help anyone for free on the community slack & github issues (spending many hours a week running & debugging others projects just to help out, and potentially find issues that can be fixed in either core, or nsv itself) - so a bit surprised I've never seen your question come up in our slack about `$modal.close`.

> In comparison to just using notepad, and vanilla Java for Android development, nativescript in its current state is worse, much worse. I very much regret going for it in a large app project, but can't get out of the trap now :(.

If you truly believe that - good luck!

**Reply by Rho-bur** [reply]
I too, have built a quite big app using NS-Vue, mostly with versions 5 to 6 included.
When I was closed to submit it to an extended beta testing, NS7 came and I thought I should upgrade sooner than later.
It was a bit challenging (I was using some older plugins I had to convert to NS7 myself) but in the end I managed it (with a considerable time spending), just a few minor things left to be ironed out currently.
This doesn't make me change my mind so I'll still choose NS on my next mobile project although I have some knowledge of Java and Swift.
Considering all and hoping the new team of maintainers will do sensibly better than the previous teams (although they have to be commended for creating this stack) I think NS has a bright future making app development less costly that having to build two of them (native Android and IoS) every time.
I second the ones wishing for better docs and faster plugin updating (although I wouldn't mind getting my hands dirty on those).
Still, NS comes free so there cannot be a support like those found (and paid for) in commercial software.
As a (serious) joke, my main discontent comes from NS leaning too much on TS (which I refuse to learn since I do not see an overwhelming plus for this) and making us, vanilla JS guys feel like a minority.
@baybal I think you should not loose the time you invested in learning it and keep up the faith, I am sure it'll be rewarded.

**Comment by thond1st** [comment]
Hi I am Anthony,
We currently have a Vue project and the team used Vuetify. We are currently integrating Nativescript-vue to the existing app for code sharing and we ran into some issues using vue-cli-plugin-nativescript-vue. I think it is outdated or doesn't support NS 7 nor NS 8 and also VueCLI 4.x.
Issues ran into and got fixed are:
1. When running the command "vue add vue-cli-plugin-nativescript-vue" into existing project. I got an eslint (Unnecessary escape character: \/ no-useless-escape) error in webpackconfig.js (got the same file as this one https://github.com/tralves/ns-vue-codeshare-demo/blob/master/webpack.config.js). I fixed it by adding this (//eslint-disable-next-line) on top of each line that have issues. But when creating this post. I checked again my webpack.config.js. It is different now and probably there was a command that I made that changed it.
2. When combining the web (<template web>) and native (<template native>) template inside the App.vue file. The rendered view is not correct but if I remove the native template the rendered view is correct. To fix it I just created App.web.vue and App.native.vue(Hope it is and acceptable fix) and just updated main.js and main.native.js files accordingly. (Note: command run was "npm run serve:web")

After these fixes the rendred view components are almost rendered correctly aside from some difference in font and some icon colors.

Then the problem comes when running "npm run preview:android" for testing but it flags deprecated error but still continued and it asked to install webpack cli. So I did and when done I just got whitescreen on my device using the Playground apps.
These are mainly the package versions:
"dependencies": {
    "axios": "^0.21.1",
    "core-js": "^3.6.5",
    "lodash": "^4.17.21",
    "moment": "^2.29.1",
    "nativescript-vue": "^2.5.0-alpha.3",
    "nativescript-vue-navigator": "^0.2.0",
    "tns-core-modules": "^6.3.2",
    "vue": "^2.6.11",
    "vue-router": "^3.2.0",
    "vuejs-paginate": "^2.1.0",
    "vuetify": "^2.4.0",
    "vuex": "^3.6.2"
  },
  "devDependencies": {
    "@babel/core": "^7.5.5",
    "@babel/preset-env": "^7.5.5",
    "@babel/traverse": "^7.5.5",
    "@vue/cli-plugin-babel": "^4.5.0",
    "@vue/cli-plugin-eslint": "^4.5.0",
    "@vue/cli-plugin-router": "^4.5.11",
    "@vue/cli-service": "^4.5.0",
    "babel-eslint": "^10.1.0",
    "babel-loader": "^8.0.6",
    "eslint": "^6.7.2",
    "eslint-plugin-vue": "^6.2.2",
    "nativescript-dev-webpack": "^1.4.0",
    "nativescript-vue-template-compiler": "^2.5.0-alpha.3",
    "nativescript-worker-loader": "~0.9.5",
    "node-sass": "^4.12.0",
    "rimraf": "^2.6.3",
    "sass": "^1.32.0",
    "sass-loader": "^10.0.0",
    "string-replace-loader": "^2.2.0",
    "tns-android": "6.3.1",
    "vue-cli-plugin-nativescript-vue": "~0.3.1",
    "vue-cli-plugin-vuetify": "^2.1.0",
    "vue-template-compiler": "^2.6.11",
    "vuetify-loader": "^1.7.0",
    "webpack-cli": "^4.6.0"
  },
  
Version checks in command line:

$> vue --version
@vue/cli 4.5.11

$> ns -v
You are using the deprecated nsconfig.json file. Just be aware that NativeScript now has an improved nativescript.config.(js|ts) file for when you're ready to upgrade this project.
8.0.1

Will post further updates here but if someone was able to integrate with a new Vue+Vuetify project. Hope you can share your process and fixes.

Btw. I can't seem to join the slack channel. It was asking for an @*nativescript*.org email.

Hoping that Nativescript, Nativescript-vue, and vue-cli-plugin-nativescript-vue will live on and flourish.
